Remove notify option NOTIFY_OPT_LOOPBACK in WebKit

pvollan opened the pull request:
#### e57afcae14d0fb780c8a2d8e74c63b2c8a8a7c6c
<pre>
Remove notify option NOTIFY_OPT_LOOPBACK in WebKit
<a href="https://bugs.webkit.org/show_bug.cgi?id=272977">https://bugs.webkit.org/show_bug.cgi?id=272977</a>
<a href="https://rdar.apple.com/126484077">rdar://126484077</a>

Reviewed by Sihui Liu.

Remove notify option NOTIFY_OPT_LOOPBACK in WebKit, since this feature is now enabled with an
entitlement. This patch effectively reverts &lt;<a href="https://commits.webkit.org/275186@main">https://commits.webkit.org/275186@main</a>&gt;. The
NOTIFY_OPT_DISPATCH applies regardless of whether NOTIFY_BLOCKING is enabled.

* Source/WebKit/Shared/AuxiliaryProcess.h:
* Source/WebKit/Shared/Cocoa/AuxiliaryProcessCocoa.mm:
(WebKit::AuxiliaryProcess::setNotifyOptions):
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceMain.mm:
(WebKit::XPCServiceEventHandler):
(WebKit::XPCServiceMain):
(WebKit::shouldRestrictNotifyd): Deleted.
(WebKit::setNotifyOptions): Deleted.
* Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm:
(WebKit::applySandbox):
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:

Canonical link: <a href="https://commits.webkit.org/278639@main">https://commits.webkit.org/278639@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9e8f10623f81c6dabb60c78223b79b5e33319cdc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/51167 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30469 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3502 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/54424 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/1857 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/36765 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1531 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/41674 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/53266 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/28102 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/44101 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22791 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/25424 "Passed tests") | | [⏳ 🛠 wpe-skia ](https://ews-build.webkit.org/#/builders/WPE-Skia-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/47404 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/1430 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/56020 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/26277 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/1321 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/49069 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/27525 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/44168 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/48208 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/memory-pressure (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11194 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/28406 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/27256 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai commented on Source/WebKit/Shared/Cocoa/AuxiliaryProcessCocoa.mm:
```diff
@@ -321,5 +322,17 @@ static void initializeTimerCoalescingPolicy()
+    if (!opts)
+        return;
+    notify_set_options(opts);
+#endif
```
It looks like this can be simplified as:
```
#if !ENABLE(NOTIFY_BLOCKING) && ENABLE(NOTIFY_FILTERING)
    notify_set_options(NOTIFY_OPT_DISPATCH | NOTIFY_OPT_REGEN | NOTIFY_OPT_FILTERED)
#endif
```

pvollan replied:
Ah, nice catch, fixed in latest patch!

Thanks for reviewing!

szewai commented on Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm:
```diff
@@ -504,6 +504,8 @@ static bool tryApplyCachedSandbox(const SandboxInfo& info)
     ASSERT(static_cast<void *>(sandboxDataPtr + profile.size) <= static_cast<void *>(cachedSandboxContents.data() + cachedSandboxContents.size()));
     profile.data = sandboxDataPtr;
 
+    AuxiliaryProcess::setNotifyOptions();
```
It seems this should be moved to before `return true;`, otherwise `setNotifyOptions()` can be invoked twice in `static bool applySandbox(...)`.

pvollan replied:
That is a good point. I have changed this to only call `setNotifyOptions` once.

szewai commented on Source/WebKit/Shared/Cocoa/AuxiliaryProcessCocoa.mm:
```diff
@@ -335,5 +336,13 @@ static void disableDowngradeToLayoutManager()
+void AuxiliaryProcess::setNotifyOptions()
+{
+#if ENABLE(NOTIFY_BLOCKING)
+    notify_set_options(NOTIFY_OPT_DISPATCH);
```
The commit message says "We only
need to set notify options the original way on macOS versions that do not have the feature
ENABLE(NOTIFY_BLOCKING)." but here we set the options when ENABLE(NOTIFY_BLOCKING). Is this intended change?

pvollan replied:
Ah, good catch! The change is intended, since we always would like to set the dispatch option. I will fix the commit message.

pvollan committed:
Remove notify option NOTIFY_OPT_LOOPBACK in WebKit
https://bugs.webkit.org/show_bug.cgi?id=272977
rdar://126484077

Reviewed by Sihui Liu.

Remove notify option NOTIFY_OPT_LOOPBACK in WebKit, since this feature is now enabled with an
entitlement. This patch effectively reverts <https://commits.webkit.org/275186@main>. The
NOTIFY_OPT_DISPATCH applies regardless of whether NOTIFY_BLOCKING is enabled.

* Source/WebKit/Shared/AuxiliaryProcess.h:
* Source/WebKit/Shared/Cocoa/AuxiliaryProcessCocoa.mm:
(WebKit::AuxiliaryProcess::setNotifyOptions):
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceMain.mm:
(WebKit::XPCServiceEventHandler):
(WebKit::XPCServiceMain):
(WebKit::shouldRestrictNotifyd): Deleted.
(WebKit::setNotifyOptions): Deleted.
* Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm:
(WebKit::applySandbox):
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:

Canonical link: https://commits.webkit.org/278639@main
